<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Generate a Manhattan Plot for GWAS Data — manhattan_plot • GWAScopeR</title><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="Generate a Manhattan Plot for GWAS Data — manhattan_plot"><meta property="og:description" content="This function generates a Manhattan plot for visualizing the results of a Genome-Wide Association Study (GWAS).
  The plot displays the -log10 of the p-values for each SNP, with horizontal threshold lines indicating significance levels.
  Vertical lines can be added to mark the top Quantitative Trait Nucleotides (QTNs) or user-provided QTN positions."><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">
    

    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">GWAScopeR</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.1.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../reference/index.html">Reference</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"></ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Generate a Manhattan Plot for GWAS Data</h1>
    
    <div class="hidden name"><code>manhattan_plot.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>This function generates a Manhattan plot for visualizing the results of a Genome-Wide Association Study (GWAS).
  The plot displays the -log10 of the p-values for each SNP, with horizontal threshold lines indicating significance levels.
  Vertical lines can be added to mark the top Quantitative Trait Nucleotides (QTNs) or user-provided QTN positions.</p>
    </div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">manhattan_plot</span><span class="op">(</span><span class="va">map_data</span>, pval_vector <span class="op">=</span> <span class="cn">NULL</span>, gwas_data <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>                threshold_type <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"fixed"</span><span class="op">)</span>, alpha <span class="op">=</span> <span class="fl">0.01</span>,</span>
<span>                show_threshold_labels <span class="op">=</span> <span class="cn">TRUE</span>, num_qtns <span class="op">=</span> <span class="fl">10</span>,</span>
<span>                user_qtn_positions <span class="op">=</span> <span class="cn">NULL</span><span class="op">)</span></span></code></pre></div>
    </div>

    <div id="arguments">
    <h2>Arguments</h2>
    <dl><dt>map_data</dt>
<dd><p>A data frame containing SNP mapping information with columns `SNP`, `Chromosome`, and `Position`.
    The rows correspond to individual SNPs.</p></dd>

  <dt>pval_vector</dt>
<dd><p>A numeric vector of p-values corresponding to the SNPs in `map_data`. The length of `pval_vector` must
    match the number of rows in `map_data`. Default is `NULL`.</p></dd>

  <dt>gwas_data</dt>
<dd><p>A data frame containing GWAS results, including a column with SNP identifiers and p-values. If provided,
    this will be used instead of `pval_vector`. The column name for p-values should be `P_Value`. Default is `NULL`.</p></dd>

  <dt>threshold_type</dt>
<dd><p>A character vector specifying the type of threshold to use for significance lines. Available options:
    "fixed", "bonferroni", "inflation", "fdr". Default is `c("fixed")`.</p></dd>

  <dt>alpha</dt>
<dd><p>A numeric value representing the significance threshold for the p-values. Default is `0.01`.</p></dd>

  <dt>show_threshold_labels</dt>
<dd><p>A logical value indicating whether threshold labels should be shown on the plot. Default is `TRUE`.</p></dd>

  <dt>num_qtns</dt>
<dd><p>A numeric value indicating the number of top Quantitative Trait Nucleotides (QTNs) to highlight on the plot.
    Default is `10`.</p></dd>

  <dt>user_qtn_positions</dt>
<dd><p>A numeric vector of user-defined QTN positions to highlight on the plot. Default is `NULL`.</p></dd>

</dl></div>
    <div id="details">
    <h2>Details</h2>
    <p>This function generates a Manhattan plot where the x-axis represents the SNPs along the chromosomes, and the y-axis
  represents the -log10 of the p-values. The function allows you to specify multiple threshold types to display
  significance lines, including Bonferroni correction, genomic inflation, false discovery rate (FDR), and a fixed alpha value.
  The plot can also highlight the top SNPs based on their p-values or user-defined positions.</p>
    </div>
    <div id="value">
    <h2>Value</h2>
    

<p>A `ggplot2` object representing the Manhattan plot.</p>
    </div>
    <div id="references">
    <h2>References</h2>
    


    </div>
    <div id="author">
    <h2>Author</h2>
    <p>Michelle Marcano-Delgado</p>
    </div>
    <div id="note">
    <h2>Note</h2>
    


    </div>
    <div id="see-also">
    <h2>See also</h2>
    <div class="dont-index"><p><code><a href="varPCs.html">varPCs</a></code>, <code><a href="https://rdrr.io/r/stats/prcomp.html" class="external-link">prcomp</a></code>, <code><a href="https://rdrr.io/r/stats/p.adjust.html" class="external-link">p.adjust</a></code>, <code><a href="https://rdrr.io/r/stats/Chisquare.html" class="external-link">qchisq</a></code></p></div>
    </div>

    <div id="ref-examples">
    <h2>Examples</h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span>  <span class="co"># Example 1: Generate Manhattan plot with p-values and GWAS data</span></span></span>
<span class="r-in"><span>  <span class="fu">manhattan_plot</span><span class="op">(</span><span class="va">map_data</span>, pval_vector <span class="op">=</span> <span class="va">pval_vector</span>, threshold_type <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"fixed"</span>, <span class="st">"bonferroni"</span><span class="op">)</span>, alpha <span class="op">=</span> <span class="fl">0.01</span><span class="op">)</span></span></span>
<span class="r-err co"><span class="r-pr">#&gt;</span> <span class="error">Error in manhattan_plot(map_data, pval_vector = pval_vector, threshold_type = c("fixed",     "bonferroni"), alpha = 0.01):</span> object 'pval_vector' not found</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span>  <span class="co"># Example 2: Generate Manhattan plot with custom top QTN positions</span></span></span>
<span class="r-in"><span>  <span class="fu">manhattan_plot</span><span class="op">(</span><span class="va">map_data</span>, gwas_data <span class="op">=</span> <span class="va">gwas_data</span>, num_qtns <span class="op">=</span> <span class="fl">5</span>, user_qtn_positions <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">100</span>, <span class="fl">200</span>, <span class="fl">300</span><span class="op">)</span><span class="op">)</span></span></span>
<span class="r-err co"><span class="r-pr">#&gt;</span> <span class="error">Error in manhattan_plot(map_data, gwas_data = gwas_data, num_qtns = 5,     user_qtn_positions = c(100, 200, 300)):</span> object 'gwas_data' not found</span>
</code></pre></div>
    </div>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by Marcano-Delgado Michelle.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

      </footer></div>

  


  

  </body></html>

